src/EditorFeatures/Test/Diagnostics/DiagnosticProviderTestUtilities.cs

```diff
@@ -49,7 +49,7 @@ private static IEnumerable<Diagnostic> GetDiagnostics(DiagnosticAnalyzer analyze
                     syntaxNodeAnalyzerService: nodeInBodyAnalyzerService,
                     hostDiagnosticUpdateSource: exceptionDiagnosticsSource,
                     cancellationToken: CancellationToken.None, 
-                    testOnly_DonotCatchAnalyzerExceptions: donotCatchAnalyzerExceptions);
+                    testOnly_RethrowAnalyzerException: donotCatchAnalyzerExceptions);
```

Comment:
@mavasani Regarding (2), I don't think that's good enough. If you want a dump then a non-fatal watson is the solution. Bringing down the compiler should never be acceptable. The only reason we crash in the compiler is that exceptions have the possibility of generating silent bad code, which is even worse than crashing.

Analyzer exceptions are always 100% recoverable by the compiler and should never bring it down, as the compiler is a critical piece of infrastructure.